### PR TITLE
Incorporate steppi's review comments, with minor tweaks

### PR DESCRIPTION
**Pull request by ragibson**
These are @steppi's reviews suggestions, with the following (very minor) tweaks.
- In the comment added in `scipy/stats/_levy_stable/c_src/levyst.c`, I changed the note about alpha = 0 having a different form (it's actually at alpha = 1)
- I tweaked the suggested lambda expression replacements in `scipy/stats/_levy_stable_distn.py`'s `_fitstart_S1()` since the syntax was slightly off.

All tests still pass:
```python
$ python runtests.py -v -m full -t scipy/stats/tests/test_distributions.py::TestLevyStable
Building, see build.log...
Build OK (0:00:03.504291 elapsed)
================================================ test session starts =================================================
platform linux -- Python 3.9.6, pytest-6.2.4, py-1.10.0, pluggy-0.13.1 -- /usr/bin/python3.9
cachedir: .pytest_cache
rootdir: /home/ryan/PycharmProjects/scipy_stable/scipy, configfile: pytest.ini
collected 12 items                                                                                                   

scipy/stats/tests/test_distributions.py::TestLevyStable::test_rvs PASSED                                       [  8%]
scipy/stats/tests/test_distributions.py::TestLevyStable::test_fit PASSED                                       [ 16%]
scipy/stats/tests/test_distributions.py::TestLevyStable::test_pdf_nolan_samples PASSED                         [ 25%]
scipy/stats/tests/test_distributions.py::TestLevyStable::test_cdf_nolan_samples PASSED                         [ 33%]
scipy/stats/tests/test_distributions.py::TestLevyStable::test_location_scale PASSED                            [ 41%]
scipy/stats/tests/test_distributions.py::TestLevyStable::test_pdf_alpha_equals_one_beta_non_zero PASSED        [ 50%]
scipy/stats/tests/test_distributions.py::TestLevyStable::test_stats PASSED                                     [ 58%]
scipy/stats/tests/test_distributions.py::TestLevyStable::test_rvs_alpha1[0.5] PASSED                           [ 66%]
scipy/stats/tests/test_distributions.py::TestLevyStable::test_rvs_alpha1[1] PASSED                             [ 75%]
scipy/stats/tests/test_distributions.py::TestLevyStable::test_distribution_outside_support[0.25] PASSED        [ 83%]
scipy/stats/tests/test_distributions.py::TestLevyStable::test_distribution_outside_support[0.5] PASSED         [ 91%]
scipy/stats/tests/test_distributions.py::TestLevyStable::test_distribution_outside_support[0.75] PASSED        [100%]

================================================ 12 passed in 45.60s =================================================
```

Two comments from the review still need to be addressed, which I'll follow up on in the main PR.

### CODE REVIEW

File: scipy/stats/_levy_stable_distn.py [hunk 1]
```diff
@@ -572,22 +572,23 @@ def _fitstart_S1(data):
     psi_2 = interpolate.interp2d(
         nu_beta_range, nu_alpha_range, beta_table, kind="linear"
     )
-    psi_2_1 = lambda nu_beta, nu_alpha: (
-        psi_2(nu_beta, nu_alpha) if nu_beta > 0 else -psi_2(-nu_beta, nu_alpha)
-    )
+
+    def psi_2_1(nu_beta, nu_alpha):
+        return psi_2(nu_beta, nu_alpha) if nu_beta > 0 else -psi_2(-nu_beta, nu_alpha)
```
Comment (steppi) [inline scipy/stats/_levy_stable_distn.py:577]: Haha. Wow, I should really pay attention better. I thought I was being clever by doing a regex find and replace on this block and and didn't even check that the result was sensible. Sorry about that.

Comment (ragibson) [reply scipy/stats/_levy_stable_distn.py:577]: haha that's okay! I actually went and initially checked if you had just used some esoteric syntax I'd never seen.

**Comment by steppi** [comment]
Awesome. Only one thing that's missing. Do you know if it was a deliberate choice to use `interpolate.InterpolatedUnivariateSpline` in the below part for the cdf and ignore `fft_interpolation_kind`?

https://github.com/bsdz/scipy/blob/21b3ba111c062b49218d6456f8ae010a104e4809/scipy/stats/_levy_stable_distn.py#L1060

I saw that for the pdf it instead uses `interpolate.interp1d`, and makes use of `fft_interpolation_kind`. 

https://github.com/bsdz/scipy/blob/21b3ba111c062b49218d6456f8ae010a104e4809/scipy/stats/_levy_stable_distn.py#L936

I think we should either make them consistent, or put something in the docs explaining why `fft_interpolation_kind` can't be set for cdf calculations with fft. I'm not sure if the constraint that the CDF is non-decreasing impacted the choice of interpolation method or something. I guess the ideal choice would be something like [Isotonic Regression](https://en.wikipedia.org/wiki/Isotonic_regression), but it's not implemented in scipy and would be pretty slow.

**Review by steppi — APPROVED** [review]
Great. I think everything's good here. Glad you caught the typo in the `levyst.c` header.

**Pull request merged**